File: DataBindingStudy/app/src/main/java/com/jie/databinding/EventBindingActivity.java
package com.jie.databinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;
import android.text.Editable;
import android.widget.Toast;

import com.jie.databinding.databinding.ActivityEventBindingBinding;
import com.jie.databinding.model.User;

import java.lang.ref.WeakReference;

public class EventBindingActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityEventBindingBinding eventBindingBinding = DataBindingUtil.setContentView(this, R.layout.activity_event_binding);
        setTitle("event binding");
        User userInfo = new User("Jack", "123456");
        eventBindingBinding.setUserInfo(userInfo);
        UserPresenter presenter = new UserPresenter(eventBindingBinding, userInfo);
        eventBindingBinding.setUserPresenter(presenter);
    }

    public class UserPresenter {
        private ActivityEventBindingBinding mEventBindingBinding;
        private User mUser;

        public UserPresenter(ActivityEventBindingBinding eventBindingBinding, User user) {
            mEventBindingBinding = eventBindingBinding;
            mUser = user;
        }

        public void onUserNameClick(User user) {
            Toast.makeText(EventBindingActivity.this, "用户名:" + user.getName(), Toast.LENGTH_LONG).show();
        }

        public void afterTextChanged(Editable e) {
            mUser.setName(e.toString());
            mEventBindingBinding.setUserInfo(mUser);
        }

        public void afterUserPwdChanged(Editable e) {
            mUser.setPwd(e.toString());
            mEventBindingBinding.setUserInfo(mUser);
        }
    }
}

File: DataBindingStudy/app/src/main/java/com/jie/databinding/ui/activity/EventBindingActivity.kt
package com.jie.databinding.ui.activity

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import com.jie.databinding.R
import com.jie.databinding.databinding.ActivityEventBindingBinding
import com.jie.databinding.model.UserEntity

class EventBindingActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val eventBinding = DataBindingUtil.setContentView<ActivityEventBindingBinding>(
            this,
            R.layout.activity_event_binding
        )
        var userInfo: UserEntity = UserEntity("Jack", "123456")
        eventBinding.userInfo = userInfo
        UserPresenter(this, userInfo, eventBinding).apply {
            eventBinding.userPresenter = this
        }
        title = "event binding study"
    }

    inner class UserPresenter constructor(
        val context: Context,
        var userInfo: UserEntity,
        val eventBinding: ActivityEventBindingBinding) {

        fun onUserNameClick(userInfo: UserEntity) {
            Toast.makeText(context, userInfo.name, Toast.LENGTH_SHORT).show()
        }

        fun afterTextChanged(editable: Editable) {
            userInfo.name = editable.toString()
            eventBinding.userInfo = userInfo
        }

        fun afterUserPwdChanged(editable: Editable) {
            userInfo.pwd = editable.toString()
            eventBinding.userInfo = userInfo
        }
    }
}